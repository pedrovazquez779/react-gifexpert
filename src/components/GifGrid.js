import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);
    
    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && <p>Cargando...</p>}

            <div className='card-grid'>
                {
                    images.map((img) => ( // estos parentesis estan haciendo un return implicito
                        <GifGridItem
                            key={img.id}
                            {...img} // pasa todas las propiedades del objeto al componente
                        />
                    ))
                }
            </div>
        </>
    )
}
