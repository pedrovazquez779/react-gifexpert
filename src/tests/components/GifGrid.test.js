import React from 'react';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Probando el GifGrid', () => {

    const category = 'One Punch';

    test('debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();

    })

    test('debe mostrarse correctamente los items cuando se cargan las imagenes con useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa',
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        })
        const wrapper = shallow(<GifGrid category={category} />)
        
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    })

})
