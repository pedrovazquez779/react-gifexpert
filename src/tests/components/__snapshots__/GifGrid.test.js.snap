// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el GifGrid debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
